src/main/scala/io/qbeast/spark/delta/IndexStatusBuilder.scala

```diff
@@ -35,41 +37,66 @@ private[delta] class IndexStatusBuilder(
     qbeastSnapshot.loadRevisionBlocks(revision.revisionID)
 
   def build(): IndexStatus = {
+    val cubeStatus =
+      if (isStaging(revision)) stagingCubeStatuses
+      else buildCubesStatuses
+
     IndexStatus(
       revision = revision,
       replicatedSet = replicatedSet,
       announcedSet = announcedSet,
-      cubesStatuses = buildCubesStatuses)
+      cubesStatuses = cubeStatus)
+  }
+
+  def stagingCubeStatuses: SortedMap[CubeId, CubeStatus] = {
+    val root = revision.createCubeIdRoot()
+    val maxWeight = Weight.MaxValue
+    val blocks = revisionFiles
+      .collect()
+      .map(addFile =>
+        QbeastBlock(
+          addFile.path,
+          revision.revisionID,
+          Weight.MinValue,
+          maxWeight,
+          FLOODED,
+          0,
+          addFile.size,
+          addFile.modificationTime))
+      .toIndexedSeq
+
+    SortedMap(root -> CubeStatus(root, maxWeight, maxWeight.fraction, blocks))
   }
 
   /**
    * Returns the index state for the given space revision
    * @return Dataset containing cube information
    */
   def buildCubesStatuses: SortedMap[CubeId, CubeStatus] = {
-
     val spark = SparkSession.active
+    import spark.implicits._
```

Comment:
If we didn't change the behaviour for this method, I think it's best to preserve the original state 🙈